modules/rest-api/pages/rest-bucket-create.adoc

```diff
@@ -620,6 +626,118 @@ curl -v -X POST http://10.143.201.101:8091/pools/default/buckets/testBucket \
 If successful, the call returns a `200 OK` notification.
 No object is returned.
 
+[#historyretentioncollectiondefault]
+=== historyRetentionCollectionDefault
+
+Whether the default, for all collections within this bucket, is that a record of changes made to documents within the collection should be created and written to disk continuously.
+The value can be either `true` (the default) or `false`.
+
+This parameter is ignored unless the value of `storageBackend` is specified as `magma`.
+This parameter is also ignored unless a positive value is specified for either `historyRetentionSeconds` or `historyRetentionBytes`, or both.
+
+[#example-flushenabled-create]
+==== Example: Enable historyRetentionCollectionDefault, when Creating
+
+The following example creates a bucket, specifies its storage as _magma_, and specifies that a record of changes made to collections within the bucket should _not_ be made.
+
+----
+curl -X POST http://localhost:8091/pools/default/buckets \
+-u Administrator:password \
+-d name=testBucket \
+-d ramQuota=1100 \
+-d storageBackend=magma \
+-d historyRetentionCollectionDefault=false
+----
+
+Success returns `202 Accepted`.
+
+[#example-flushenabled-edit]
+==== Example: Modify historyRetentionCollectionDefault, when Editing
+
+The following example modifies the value of `historyRetentionCollectionDefault` for the existing bucket `testBucket`.
+
+----
+curl -v -X POST http://localhost:8091/pools/default/buckets/testBucket \
+-u Administrator:password \
+-d historyRetentionCollectionDefault=true
+----
+
+Success returns `200 OK`.
+Note, however, that this call only results in a change-history being written to disk if `storageBackend` was specified as `magma` at bucket-creation, and if a positive value is specified (either prior to or after this call) for either `historyRetentionSeconds` or `historyRetentionBytes`, or for both.
+
+[#historyretentionbytes]
+=== historyRetentionBytes
+
+Specifies the maximum size, in bytes, of the change-history that is written to disk for all collections in this bucket when the value of `historyRetentionCollectionDefault` is `true`.
```

Comment:
I wonder if we need to clarify that each replica multiplies the amount required.

E.g. 2 replicas and set a value of 100MB - that really becomes 200MB

Reply:
I've written: Note that each replica configured for the bucket maintains a copy of the change-history: therefore, if two replicas are configured, and the specified maximum size is 1,000,000 bytes, the total size used for the change-history across the cluster becomes 3,000,000 bytes.